### PR TITLE
Documented constraint library and prop.create

### PR DESCRIPTION
**Pull request by adamnejm**

### CODE REVIEW

**Comment by thegrb93** [comment]
Thanks. That's really good. I'll look later tonight.

File: lua/starfall/libs_sv/constraint.lua [hunk 1]
```diff
@@ -145,14 +156,22 @@ function constraint_library.axis(e1, e2, bone1, bone2, v1, v2, force_lim, torque
 	end
 end
 
---- Ballsocket two entities
+--- Ballsocket two entities together. For more options, see constraint.ballsocketadv
+-- @param e1 The first entity
+-- @param e2 The second entity
+-- @param bone1 Number bone of the first entity
+-- @param bone2 Number bone of the second entity
+-- @param pos Position of the joint, relative to the second entity
```
Comment (thegrb93) [inline lua/starfall/libs_sv/constraint.lua:164]: Need to verify this is local to second entity

File: lua/starfall/libs_sv/constraint.lua [hunk 2]
```diff
@@ -174,7 +193,20 @@ function constraint_library.ballsocket(e1, e2, bone1, bone2, v1, force_lim, torq
...
+--- Ballsocket two entities together with more options
+-- @param e1 The first entity
+-- @param e2 The second entity
+-- @param bone1 Number bone of the first entity
+-- @param bone2 Number bone of the second entity
+-- @param v1 Position on the first entity, in its local space coordinates
+-- @param v2 Position on the second entity, in its local space coordinates
+-- @param force_lim Amount of force until it breaks, 0 = Unbreakable
+-- @param torque_lim Amount of torque until it breaks, 0 = Unbreakable
+-- @param minv Vector defining minimum rotation angle, local to the constraint
```
Comment (thegrb93) [inline lua/starfall/libs_sv/constraint.lua:205]: I think this angle limits on world axes. Need to verify.

File: lua/starfall/libs_sv/constraint.lua [hunk 3]
```diff
@@ -113,6 +113,17 @@ function constraint_library.weld(e1, e2, bone1, bone2, force_lim, nocollide)
...
+-- @param e2 The second entity
+-- @param bone1 Number bone of the first entity
+-- @param bone2 Number bone of the second entity
+-- @param v1 Position on the first entity, in its local space coordinates
+-- @param v2 Position on the second entity, in its local space coordinates
+-- @param force_lim Amount of force until it breaks, 0 = Unbreakable
+-- @param torque_lim Amount of torque until it breaks, 0 = Unbreakable
+-- @param friction Friction of the constraint
+-- @param nocollide Bool whether or not to nocollide the two entities
+-- @param laxis Local axis, if set, then position on the second entity (v2 argument) will not be used in the final constraint, however it's still required
```
Comment (thegrb93) [inline lua/starfall/libs_sv/constraint.lua:126]: This sets the axis normal in the coordinates of the first entity I think. Need to verify

File: lua/starfall/libs_sv/constraint.lua (hunk 2 above)
Comment (adamnejm) [reply lua/starfall/libs_sv/constraint.lua:205]: World axis is correct

File: lua/starfall/libs_sv/constraint.lua (hunk 1 above)
Comment (adamnejm) [reply lua/starfall/libs_sv/constraint.lua:164]: It is indeed relative to the second entity

File: lua/starfall/libs_sv/constraint.lua (hunk 3 above)
Comment (adamnejm) [reply lua/starfall/libs_sv/constraint.lua:126]: You're correct, i will additionally simplify the description cuz it's a mess

**Pull request merged**